src/classes/BDI_MigrationMappingUtility.cls

```diff
@@ -0,0 +1,337 @@
+
+        // If one exists, reset instance with newly added field
+        if (exists != null) {
+            importRelatedField = new BDI_HelpTextCMT.ImportStatusAndLookupParts(exists, dataImportFieldName);
+        }
+
+        return importRelatedField;
+    }
+
+    /**
+     * @description Queries for default Object Mappings and builds a map of DeveloperName to Object Mapping record
+     * Used to check for existance of Object Mapping record without loop
+     * @param objectMappingSetId Id of the default Object Mapping Set
+     * @return Collection of Object Mapping DeveloperName to Object Mapping record
+     */
+    private Map<String, Data_Import_Object_Mapping__mdt> getDefaultObjectMappings(Id objectMappingSetId) {
+        List<Data_Import_Object_Mapping__mdt> existingObjectMappings = [SELECT Id,
+                                                                            Label,
+                                                                            DeveloperName,
+                                                                            Imported_Record_Field_Name__c,
+                                                                            Imported_Record_Status_Field_Name__c,
+                                                                            Object_API_Name__c,
+                                                                            Predecessor__c,
+                                                                            Relationship_Field__c,
+                                                                            Relationship_To_Predecessor__c,
+                                                                            Is_Deleted__c
+                                                                        FROM Data_Import_Object_Mapping__mdt
+                                                                        WHERE Data_Import_Object_Mapping_Set__c = :objectMappingSetId];
+        Map<String, Data_Import_Object_Mapping__mdt> diObjectNameToObjectMappingMap = new Map<String, Data_Import_Object_Mapping__mdt>();
+
+        for (Data_Import_Object_Mapping__mdt objectMapping : existingObjectMappings) {
+            diObjectNameToObjectMappingMap.put(objectMapping.DeveloperName, objectMapping);
+        }
+
+        return diObjectNameToObjectMappingMap;
+    }
+
+    /**
+     * @description Queries for default Field Mappings and builds a map of DeveloperName to Field Mapping record
+     * Used to check for existance of Field Mapping record without loop
+     * @param fieldMappingSetId Id of the default Field Mapping Set
+     * @return Collection of Field Mapping DeveloperName to Field Mapping record
+     */
+    private Map<String, Data_Import_Field_Mapping__mdt> getDefaultFieldMappings(Id fieldMappingSetId) {
+        List<Data_Import_Field_Mapping__mdt> existingFieldMappings = [SELECT Id,
```

Comment:
This isn't going to block here, but can we format these SOQL queries .... slightly less far to the right? I think there's some guidance in the style guide